### PR TITLE
Implement AssemblyLoader with error handling

### PR DESCRIPTION
**Pull request by jbrinkman**
This PR implements task 3.1 from the implementation plan:

## Changes
- Implemented `AssemblyLoader` class that safely loads .NET assemblies using `Assembly.LoadFrom` with proper error handling
- Implemented `IsolatedAssemblyLoadContext` for assembly isolation
- Added comprehensive error handling for various scenarios:
  - File not found
  - Invalid assembly format
  - Security restrictions
  - Path too long
  - Reflection type load exceptions
- Added caching of loaded assemblies for performance
- Implemented proper disposal of assembly contexts
- Fixed and updated unit tests

## Requirements Addressed
- 1.1: Loading assemblies from file paths
- 1.4: Proper error handling when assemblies cannot be loaded

## Testing
All unit tests are passing, covering various error scenarios and edge cases.

### CODE REVIEW

File: src/DotNetApiDiff/AssemblyLoading/AssemblyLoader.cs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+            {
+                loadContext.AddSearchPath(assemblyDirectory);
+            }
+            
+            // Load the assembly in the isolated context
+            var assembly = loadContext.LoadFromAssemblyPath(assemblyPath);
+            
+            // Store the assembly and load context for later use
+            _loadedAssemblies[assemblyPath] = assembly;
+            _loadContexts[assemblyPath] = loadContext;
```
[the remark this answers is not in the document]
Comment (jbrinkman) [reply src/DotNetApiDiff/AssemblyLoading/AssemblyLoader.cs:82]: Our use case does not suffer from this issue. No change needed.
